koku/api/report/aws/provider_map.py

```diff
@@ -270,6 +270,122 @@ def __init__(self, provider, report_type, schema_name, cost_type, markup_cost="m
+                            "sup_usage": Value(0, output_field=DecimalField()),
+                            "sup_markup": Value(0, output_field=DecimalField()),
+                            "sup_total": Value(0, output_field=DecimalField()),
+                            "cost_raw": Sum(
+                                Coalesce(F(self.cost_type), Value(0, output_field=DecimalField()))
+                                * Coalesce("exchange_rate", Value(1, output_field=DecimalField()))
+                            ),
+                            "cost_usage": Value(0, output_field=DecimalField()),
+                            "cost_markup": Sum(
+                                Coalesce(F(self.markup_cost), Value(0, output_field=DecimalField()))
+                                * Coalesce("exchange_rate", Value(1, output_field=DecimalField()))
+                            ),
+                            "cost_total": Sum(
+                                (
+                                    Coalesce(F(self.cost_type), Value(0, output_field=DecimalField()))
+                                    + Coalesce(F(self.markup_cost), Value(0, output_field=DecimalField()))
+                                )
+                                * Coalesce("exchange_rate", Value(1, output_field=DecimalField()))
+                            ),
+                            # the `currency_annotation` is inserted by the `annotations` property of the query-handler
+                            "cost_units": Coalesce("currency_annotation", Value("USD", output_field=CharField())),
+                            "usage": Sum("usage_amount"),
+                            "usage_units": Coalesce(Max("unit"), Value("Hrs")),
+                            "source_uuid": ArrayAgg(
+                                F("source_uuid"), filter=Q(source_uuid__isnull=False), distinct=True
+                            ),
+                            "account_alias": Max("account_alias"),
+                            "account": Max("usage_account_id"),
+                            "instance_name": Max("instance_name"),
+                            "instance_type": Max("instance_type"),
+                            "operating_system": Max("operating_system"),
+                            "region": Max("region"),
+                            "vcpu": Max("vcpu"),
+                            "memory": Max("memory"),
+                            "tags": ArrayAgg(F("tags")),
+                        },
+                        "filter": [{}],
+                        "report_type_filters": {
+                            # report specific filters
+                            "operating_system": {"field": "operating_system", "operation": "icontains"},
+                            "instance_name": {"field": "instance_name", "operation": "icontains"},
+                            "resource_id": {"field": "resource_id", "operation": "icontains"},
+                        },
```

Comment:
These can just be added to the full list of filters because the serializer is responsible for ensuring that the filters we're using on a given endpoint are correct. As an example, `az` is in the list of filters, but when you go here:
http://localhost:8000/api/cost-management/v1/reports/aws/resources/ec2-compute/?filter[az]=us-east-1
you see the validation error:
```
{
    "filter": {
        "az": [
            "Unsupported parameter or invalid value"
        ]
    }
}
```

So if you add these report specific filters to the full list above, these won't be valid on the other endpoints because of the serializer. If we just add the filters above to the big list of filters, we can undo the change in `koku/api/report/queries.py`.

Reply:
Yep, no need for `report_type_filters`. Thanks!